src/Microsoft.AspNet.Http.Features/ISession.cs

```diff
@@ -1,20 +1,20 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace Microsoft.AspNet.Http.Features
 {
     public interface ISession
     {
-        void Load();
+        Task LoadAsync();
```

Comment:
We should add a state field to indicate if the session has been loaded, and one to show if it has been modified since loaded. Right now you can't tell if you need to call LoadAsync or CommitAsync.